src/bobtail/dag.cpp

```diff
@@ -127,40 +189,44 @@ bool CDagForrest::Insert(const CSubBlock &sub_block)
             continue;
         }
         newNode->AddAncestor(ancestor);
-        merge_list.emplace(ancestor->subgraph_id);
+        merge_list.emplace(ancestor->dag_id);
         ancestor->AddDescendant(newNode);
     }
     int16_t new_id = -1;
     if (merge_list.size() > 1)
     {
-        new_id = MergeTrees(merge_list);
+        if (!MergeDags(merge_list, new_id))
+        {
+            return false;
+        }
     }
     else if (merge_list.size() == 1)
     {
         new_id = *(merge_list.begin());
     }
     else // if(merge_list.size() == 0)
     {
-        new_id = next_subgraph_id;
-        next_subgraph_id = next_subgraph_id + 1;
+        new_id = vdags.size();
+        vdags.emplace_back(new_id, newNode);
     }
-    newNode->subgraph_id = new_id;
+    newNode->dag_id = new_id;
+    vdags[new_id].Insert(newNode);
     // TODO : should insert to maintain temporal ordering not just emplace_back
-    _dag.emplace_back(newNode);
+    mapAllNodes.emplace(newNode->hash, newNode);
     return true;
 }
 
-void CDagForrest::TemporalSort()
+void CBobtailDagSet::TemporalSort()
 {
 
 }
 
-bool CDagForrest::IsTemporallySorted()
+bool CBobtailDagSet::IsTemporallySorted()
 {
     return true;
 }
 
-bool CDagForrest::IsSubgraphValid(std::set<uint256> sgHashes)
+bool CBobtailDagSet::IsSubgraphValid(std::set<uint256> sgHashes)
```

Comment:
I got the sense that you were replacing "subgraph" with "dag" thought? Is that the case? If so, shouldn't this be `IsDagValid`?

Reply:
method was removed